data-access/src/app/application-services/users/vendor-user/vendor-user.domain.ts

```diff
@@ -0,0 +1,73 @@
+import { VendorUserConverter, VendorUserDomainAdapter, VendorUserRepository } from "../../../external-dependencies/domain";
+import { DomainDataSource } from "../../../data-sources/domain-data-source";
+import { VendorUser } from "../../../domain/contexts/users/vendor-user/vendor-user";
+import { VendorUserVisa } from "../../../domain/contexts/users/vendor-user/vendor-user.visa";
+import { VendorUserData } from "../../../external-dependencies/datastore";
+import { UserUpdateInput } from "../../../external-dependencies/graphql-api";
+import { AppContext } from "../../../init/app-context-builder";
+
+export interface VendorUserDomainApi {
+  addUser() : Promise<VendorUserData>;
+  updateUser(user: UserUpdateInput) : Promise<VendorUserData>; 
+}
+type PropType = VendorUserDomainAdapter;
+type DomainType = VendorUser<PropType>;
+type RepoType = VendorUserRepository<PropType>;
+
+export class VendorUserDomainApiImpl
+  extends DomainDataSource<AppContext, VendorUserData, PropType, VendorUserVisa, DomainType, RepoType>
+  implements VendorUserDomainApi {
+
+  async updateUser(user: UserUpdateInput): Promise<VendorUserData> {
+    if (!this.context.verifiedUser || this.context.verifiedUser.openIdConfigKey !== 'AccountPortal') {
+      throw new Error('Unauthorized');
+    }
+
+    let result: VendorUserData;
+    await this.withTransaction(async (repo) => {
+      let domainObject = await repo.get(user.id);
+      if (!domainObject || domainObject.externalId !== this.context.verifiedUser.verifiedJWT.sub) {
+        throw new Error('Unauthorized');
+      }
```

Comment:
Add the missing fields from VendorUserUpdateInput to check if the field is not undefined and then call the setter on the domain object using the input field.